src/main/java/com/alvarium/sign/Ed25519Provider.java

```diff
@@ -0,0 +1,40 @@
+package com.alvarium.sign;
+
+import java.security.GeneralSecurityException;
+
+import com.alvarium.utils.Encoder;
+import com.google.crypto.tink.subtle.Ed25519Sign;
+import com.google.crypto.tink.subtle.Ed25519Verify;
+
+public class Ed25519Provider implements SignProvider {
+
+  public String sign(byte[] key, byte[] data) throws SignException {
+    
+    Ed25519Sign signer;
+
+    try {
+      signer = new Ed25519Sign(key);
+    } catch(GeneralSecurityException e) {
+      throw new SignTypeException("SHA512 not supported");
```

Comment:
Unsure why this says SHA512 specifically.

I notice also in this and the other catch blocks that the error message from the caught exception is being over-written or thrown out. I recommend not doing that. It's fine to append your own messages but in general I'd preserve visibility to the underlying exception.

Try something like:
```
public class SignException extends Exception{
    public SignException(String message, Exception e) {
        super(message, e, false, true);
    }
}
```
Where `message` is your specific message and `e` is the caught exception. Note that the above example is for `SignException` not `SignTypeException` since the latter inherits from the former. In fact if you do this, you could potentially collapse these two exception types into one with the specific message providing the differentiation.

Reply:
Regarding the first point, the docs state that the `Ed25519Sign` constructor only fails if SHA512 is not available in the message digest package of the jdk, I am guessing that's the algorithm being used to hash the content. 

I do have to say that I was a bit skeptical replacing all exceptions with my own, but your approach makes a lot of sense and I assume at the application layer we would have a stack trace of all the caught exceptions at any layer, I'll work on that and push the changes.